### DISCUSSION TITLE
Persistent uploady state

### DISCUSSION THREAD

**Discussion by TheMadKow** [rpldy/react-uploady#271] [Q&A] [answered]
Hi, Is there a way to start uploady with previous state ?
That is, save the state (say a store) and retrieve it later on.

It's useful for the user to start uploading files, leave for a while, and add files later.
In my case, the user creates email, add files, going to do some other work, and returns to this specific email to add more files. 

I didn't find a way to do that, maybe I've skipped it.
I can register for changes to get the batch state, but how to set stored value for newly created uploday instance ?

Bset,
Eliran ;)

**Comment by yoavniran** [comment, marked as answer]
Hi @TheMadKow 
From what you described I think it will be possible to achieve this simply by moving the ```<Uploady>``` declaration to a higher place in your app. 

This way, you will only have one Uploady instance and you wont need to transfer state.

Also, if the user uploads files, does more work, and then comes back to the email. why do you need to transfer the "state" from the previous uploads? Arent they already uploaded? 
If you need a way to show that info about the files uploaded, you can store the info for the item(s) that you get from a hook such as `useItemFinalizeListener` in your app's state. 

In general I dont recommend relying on Uploady's inner state for an app's UI. It's better to save the info you need into the app's state using Uploady's hooks/events

**Reply by smartlightning** [reply]
Where would I then put my custom upload button if we move the Uploady declaration higher?

**Reply by yoavniran** [reply]
you can put it anywhere as long as its in the rendering tree below the uploady component.
info is shared through Context so there's no need to place your button as a direct descendant of the Uploady component
